server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -835,432 +835,550 @@ public void deleteSnapshots(
         long repositoryDataGeneration,
         IndexVersion repositoryFormatIndexVersion,
         SnapshotDeleteListener listener
+    ) {
+        createSnapshotsDeletion(snapshotIds, repositoryDataGeneration, repositoryFormatIndexVersion, new ActionListener<>() {
+            @Override
+            public void onResponse(SnapshotsDeletion snapshotsDeletion) {
+                snapshotsDeletion.runDelete(listener);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                listener.onFailure(e);
+            }
+        });
+    }
+
+    private void createSnapshotsDeletion(
+        Collection<SnapshotId> snapshotIds,
+        long repositoryStateId,
```

Comment:
Should this be named `repositoryDataGeneration`?

Reply:
Thanks, yes, I missed that one -- 74cb165bf34